### PR TITLE
Open Join and Open Edit

### PR DESCRIPTION
**Pull request by lwjameson**
## In this PR

This PR reworks a number of aspects of creating new projects and adds new features for projects in general.

### Project Creation Dialog

* Creating a new project via the Create Project button now presents a user with a dialog to specify the parameters of the project.

image

  * This dialog requires both name and description to be filled in before Save is activated
  * The info circles will put up an tooltip explaining the feature 
  * Once created you are immediately taken to the Project's home page.

* Open Join - Open join projects allow all registered users to join your project without an invitation
* Open Edit - Allows all members of the project to add annotations to the default context

* The project creation process has been moved to a server side RPC.

* Added column is_project_default to contexts table, eliminating dependence on checking for a null name or DEFAULT_CONTEXT tag to identify.

### Project Settings updated

* You can now adjust most aspects of the project from the Settings page.

image

### Misc fixes

* Added an "Add an annotation..." placeholder to the Rich Text Editor.

image

* Updated the Profile page to be more descriptive.

image

### CODE REVIEW

**Comment by lwjameson** [comment]
@rsimon I have deployed the server requirements to recogito-staging.covecollective.org server. I can also deploy it to the Bonn staging server if that would be easier for you to review.

File: src/apps/dashboard-projects/ProjectsHome.tsx
Comment (rsimon) [inline src/apps/dashboard-projects/ProjectsHome.tsx:66]: `me` is coming in as a prop on mount, and wouldn't change I believe. Is there a need for the extra effect hook?

File: src/Types.ts
Comment (rsimon) [inline src/Types.ts:180]: +1. We used to do this via tagging. Does this mean the tag-based approach for identifying the default context is no longer needed?

File: src/apps/project-home/ProjectHome.tsx
Comment (rsimon) [inline src/apps/project-home/ProjectHome.tsx:72]: Just for my understanding: can the defaultContext ever be undefined? (Or would this point to an integrity problem in the DB?)

File: src/apps/project-settings/ProjectSettings.css
Comment (rsimon) [inline src/apps/project-settings/ProjectSettings.css:57]: We should definitely move those general styles (for the switch) into the global stylesheet to avoid repetition further down the line.

File: src/apps/project-settings/ProjectSettings.tsx
Comment (rsimon) [inline src/apps/project-settings/ProjectSettings.tsx:2]: We should definitely also refactor this at some point. Project access settings and the tag vocabulary are separate things, and we should pack them into separate components. Perhaps wait for Chelseas review first, but we should IMO put it on the list.

File: src/backend/helpers/documentHelpers.ts
Comment (rsimon) [inline src/backend/helpers/documentHelpers.ts:198]: Leftover debug log :-)

File: src/components/CreateProjectDialog/CreateProjectDialog.tsx
Comment (rsimon) [inline src/components/CreateProjectDialog/CreateProjectDialog.tsx:1]: If this is only used on the dashboard, I wonder if we should just move it into `apps/dashboard-projects/CreateProjectDialog`?

File: src/components/InfoTooltip/InfoTooltip.css
Comment (rsimon) [inline src/components/InfoTooltip/InfoTooltip.css:1]: There are already some general CSS tooltip styles in `themes/default/tooltip`. I.e. much of this might not be needed. Also to point out that the demo styles from the Radix docs are referencing color vars we don't actually have!

I believe it should actually be fine to simply remove this stylesheet altogether. As long `tooltip`, `tooltip-arrow` and `tooltip-content` classes to your tooltip, it should pick up the base styles we already have.

File: src/components/InfoTooltip/InfoTooltip.tsx
Comment (rsimon) [inline src/components/InfoTooltip/InfoTooltip.tsx:17]: Re above: I think if you replace `info-tooltip-content` with `tooltip-content` and `info-tooltip-arrow` with `tooltip-arrow`, our base styles should kick in.

**Review by rsimon — APPROVED** [review]
Looks good! Left a few comments & questions, but really mostly for tweaks and possible future refactoring.

File: src/apps/dashboard-projects/ProjectsHome.tsx
Comment (lwjameson) [reply src/apps/dashboard-projects/ProjectsHome.tsx:66]: Probably my over zealous/SPA bent where I never trust a prop on mount.

File: src/Types.ts
Comment (lwjameson) [reply src/Types.ts:180]: I believe so yes. When looking at the code, you almost always find the default context by checking if its name is null. I actually did not see anywhere where the code check the tag.  If you know of a spot let me know.

File: src/components/InfoTooltip/InfoTooltip.tsx
Comment (lwjameson) [reply src/components/InfoTooltip/InfoTooltip.tsx:17]: I will make that change.

File: src/apps/project-home/ProjectHome.tsx
Comment (lwjameson) [reply src/apps/project-home/ProjectHome.tsx:72]: Yes. If the user is not a member of a layer in the default context, then it is null as it does not pass RLS.  It hasn't been a problem I guess since that user should. not see any documents anyway, but if use a student user in Production you will see an error thrown every time they go into the project home.

File: src/components/CreateProjectDialog/CreateProjectDialog.tsx
Comment (lwjameson) [reply src/components/CreateProjectDialog/CreateProjectDialog.tsx:1]: Yeah.  I have not really always made the Only used in this app, general purpose component.  I am not sure we have been totally consistent on this.

File: src/components/InfoTooltip/InfoTooltip.css
Comment (lwjameson) [reply src/components/InfoTooltip/InfoTooltip.css:1]: I have tended to add the custom CSS.  I do believe I have changed the vars to match ours when we actually used the class, but I see I left some unused classes from the demo :-/

**Comment by lwjameson** [comment]
@rsimon I will probably need you to update the german translations before I merge.

**Pull request merged**
